### PR TITLE
remove `fix_return_consumed_capacity`, fixes #6437

### LINKED ISSUE #6437: bug: DynamoDB - PutItem default response is not the same as AWS
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

PutItem on dynamodb returns ConsumedCapacity Information.
```json
{
  "ConsumedCapacity": {
    "TableName": "put-item-response",
    "CapacityUnits": 1.0
  }
}
```

### Expected Behavior

The response should be an empty map.

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)
I have tested this against the following images:
```bash
# docker run --rm -it -p 4566:4566 -p 4510-4559:4510-4559 localstack/localstack:latest
LocalStack version: 0.13.0.4
LocalStack build date: 2021-11-24
LocalStack build git hash: f1dfb02b
```

```bash
# docker run --rm -it -p 4566:4566 -p 4510-4559:4510-4559 localstack/localstack:0.14.5
LocalStack version: 0.14.5
LocalStack build date: 2022-07-08
LocalStack build git hash: db54501d
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)
```bash
# Create table to repro the issue
awslocal dynamodb create-table \
  --table-name "put-item-response" \
  --billing-mode "PAY_PER_REQUEST" \
  --attribute-definitions \
    AttributeName=pkey,AttributeType=S \
  --key-schema \
    AttributeName=pkey,KeyType=HASH \
  --region eu-west-1

# This is the main issue, note the response on this command.
awslocal dynamodb put-item \
  --table-name "put-item-response" \
  --item '{"pkey":{"S":"test"}}'

# Cleanup resources
awslocal dynamodb delete-table \
  --table-name "put-item-response"
```

### Environment

```markdown
- OS: MacOS Monterey, ARM
- LocalStack: latest,0.15.4
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by brandonstubbs**
`fix_return_consumed_capacity` was added for issue #2049.

This seems to be the cause of issue #6437 and it doesn't solve the problem for issue #2049 from what I can tell.
If `ReturnValues` are not specified it always forces `ReturnConsumedCapacity` as `TOTAL`.

Steps to try reproduce #2049 without success:
```bash
# Create table to repro the issue
awslocal dynamodb create-table \
  --table-name "put-item-response" \
  --billing-mode "PAY_PER_REQUEST" \
  --attribute-definitions \
    AttributeName=pkey,AttributeType=S \
  --key-schema \
    AttributeName=pkey,KeyType=HASH \
  --region eu-west-1

# This works as expected, no values are returned as there is no previous item
awslocal dynamodb put-item \
  --table-name "put-item-response" \
  --item '{"pkey":{"S":"test"}}' \
  --return-values "ALL_OLD"

# This works as expected, the values for the previous item is returned.
awslocal dynamodb put-item \
  --table-name "put-item-response" \
  --item '{"pkey":{"S":"test"},"val":{"S":"foo"}}' \
  --return-values "ALL_OLD"

# Cleanup resources
awslocal dynamodb delete-table \
  --table-name "put-item-response"
```

### CODE REVIEW

**Review by giograno — CHANGES_REQUESTED** [review]
Thanks for your contribution @brandonstubbs 👍 
I did check against AWS and indeed, `ConsumedCapacity` should not be in the response.
Could you please add a simple test to this PR?
Something along this lines
```python
def test_return_values(self, dynamodb, dynamodb_client):
    aws_stack.create_dynamodb_table(
        TEST_DDB_TABLE_NAME, partition_key=PARTITION_KEY, client=dynamodb_client
    )
    table = dynamodb.Table(TEST_DDB_TABLE_NAME)
    item1 = {PARTITION_KEY: "id1"}
    item2 = {PARTITION_KEY: "id1", "data": "foobar"}

    response = table.put_item(Item=item1, ReturnValues="ALL_OLD")
    assert response["ResponseMetadata"]["HTTPStatusCode"] == 200
    assert "Attributes" not in response
    assert "ConsumedCapacity" not in response

    response = table.put_item(Item=item2, ReturnValues="ALL_OLD")
    assert response["ResponseMetadata"]["HTTPStatusCode"] == 200
    assert "ConsumedCapacity" not in response
    assert response["Attributes"] == item1

    # clean up
    table.delete()
```
You can even test it against AWS yourself with the following command:
`TEST_TARGET="AWS_CLOUD" TEST_PATH=" tests/integration/test_dynamodb.py::TestDynamoDB::test_return_values" make test`

Thank you very much!

**Comment by brandonstubbs** [comment]
@giograno no problem at all, found an existing test for the put item validation and added the additional cases that were missing.

Thank you for the prompt reply, have a great day!

File: tests/integration/test_dynamodb.py [hunk 1]
```diff
@@ -427,29 +427,50 @@ def test_return_values_in_put_item(self, dynamodb):
...
         # items which are being used to put in the table
         item1 = {PARTITION_KEY: "id1", "data": "foobar"}
+        item1b = {PARTITION_KEY: "id1", "data": "barfoo"}
         item2 = {PARTITION_KEY: "id2", "data": "foobar"}
 
         response = table.put_item(Item=item1, ReturnValues="ALL_OLD")
         # there is no data present in the table already so even if return values
         # is set to 'ALL_OLD' as there is no data it will not return any data.
-        assert not response.get("Attributes")
+        assert response["ResponseMetadata"]["HTTPStatusCode"] == 200
```
Comment (giograno) [inline tests/integration/test_dynamodb.py:436]: nitpick: we are calling the same asserts a bunch of times. To be a bit more concise, you can wrap them in an inner function.

**Review by giograno — CHANGES_REQUESTED** [review]
Thanks for the changes 👍  Just a minor nitpick in the test.
You can try to run the test against the real AWS (as I explained in a previous comment) and mark the test with `@pytest.mark.aws.validated`. Thanks again 🙏

**Comment by brandonstubbs** [comment]
Simplifies the assertions with a validation function and ran it against real AWS using `TEST_TARGET="AWS_CLOUD"`

File: tests/integration/test_dynamodb.py [hunk 2]
```diff
@@ -421,35 +421,58 @@ def test_more_than_20_global_secondary_indexes(self, dynamodb, dynamodb_client):
...
         # items which are being used to put in the table
         item1 = {PARTITION_KEY: "id1", "data": "foobar"}
+        item1b = {PARTITION_KEY: "id1", "data": "barfoo"}
         item2 = {PARTITION_KEY: "id2", "data": "foobar"}
 
         response = table.put_item(Item=item1, ReturnValues="ALL_OLD")
         # there is no data present in the table already so even if return values
         # is set to 'ALL_OLD' as there is no data it will not return any data.
-        assert not response.get("Attributes")
+        validate_response(response)
```
Comment (giograno) [inline tests/integration/test_dynamodb.py:452]:
```suggestion
        _validate_response(response)
```

File: tests/integration/test_dynamodb.py [hunk 3]
```diff
@@ -421,35 +421,58 @@ def test_more_than_20_global_secondary_indexes(self, dynamodb, dynamodb_client):
...
+        # now a previous version of data is present, so when we pass return
+        # values as 'ALL_OLD' it should give us the old attributes
+        response = table.put_item(Item=item1b, ReturnValues="ALL_OLD")
+        validate_response(response, expected={"Attributes": item1})
 
         response = table.put_item(Item=item2)
         # we do not have any same item as item2 already so when we add this by default
         # return values is set to None so no Attribute values should be returned
-        assert not response.get("Attributes")
+        validate_response(response)
```
Comment (giograno) [inline tests/integration/test_dynamodb.py:466]:
```suggestion
        _validate_response(response)
```

File: tests/integration/test_dynamodb.py [hunk 4]
```diff
@@ -421,35 +421,58 @@ def test_more_than_20_global_secondary_indexes(self, dynamodb, dynamodb_client):
...
 
         response = table.put_item(Item=item2)
         # in this case we already have item2 in the table so on this request
         # it should not return any data as return values is set to None so no
         # Attribute values should be returned
-        assert not response.get("Attributes")
+        validate_response(response)
```
Comment (giograno) [inline tests/integration/test_dynamodb.py:472]:
```suggestion
        _validate_response(response)
```

**Review by giograno — COMMENTED** [review]
Nice changes @brandonstubbs 👍  just left a really minor suggestion about naming the helping function :)

**Pull request merged**